bitmark-references/bitmark-sdk/query.md

```diff
@@ -0,0 +1,374 @@
+    .referenced(toBitmarkID: "58737de5ad68a535da6277da62d11eb3ed76ff6dd7fc2adf3c42a4096d9a2518")
+    .at(offset)
+    .to(direction: .earlier)
+    .loadAsset(true)
+    .includePending(true)
+let (txs, assets) = try Transaction.list(params: query)
+````
+
+```go
+// import bmTx "github.com/bitmark-inc/bitmark-sdk-go/tx"
+params := bmTx.NewQueryParamsBuilder().
+    OwnedBy("eZpG6Wi9SQvpDatEP7QGrx6nvzwd6s6R8DgMKgDbDY1R5bjzb9").
+    ReferencedAsset("0e0b4e3bd771811d35a23707ba6197aa1dd5937439a221eaf8e7909309e7b31b6c0e06a1001c261a099abf04c560199db898bc154cf128aa9efa5efd36030c64").
+    ReferencedBitmark("58737de5ad68a535da6277da62d11eb3ed76ff6dd7fc2adf3c42a4096d9a2518").
+    LoadAsset(true).
+    Limit(10)
+txs, referencedAssets, err := bmTx.List(params)
+```
+
+```java
+TransactionQueryBuilder builder = new TransactionQueryBuilder()
+                                .ownedBy("eZpG6Wi9SQvpDatEP7QGrx6nvzwd6s6R8DgMKgDbDY1R5bjzb9")
+                                .referencedAsset("0e0b4e3bd771811d35a23707ba6197aa1dd5937439a221eaf8e7909309e7b31b6c0e06a1001c261a099abf04c560199db898bc154cf128aa9efa5efd36030c64")
+                                .referencedBitmark("58737de5ad68a535da6277da62d11eb3ed76ff6dd7fc2adf3c42a4096d9a2518")
+                                .loadAsset(true)
+                                .limit(10);
+Transaction.list(builder, new Callback1<GetTransactionsResponse>() {
+            @Override
+            public void onSuccess(GetTransactionsResponse res) {
+
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+
+            }
+        });
+```
+
+### Query the provenance of a bitmark
+````javascript
+let transactionQueryParams = Transaction.newTransactionQueryBuilder()
+    .referencedBitmark("c8e021c1a093c32909e4d29b4624f8a5443e349a597314b7c9527ce310749121")
+    .build();
+
+let response = await Transaction.list(transactionQueryParams);
+````
+
+````swift
+let query = try Transaction.newTransactionQueryParams()
+    .referenced(toBitmarkID: "58737de5ad68a535da6277da62d11eb3ed76ff6dd7fc2adf3c42a4096d9a2518")
+let (txs, _) = try Transaction.list(params: query)
+````
+
+```go
+// import bmTx "github.com/bitmark-inc/bitmark-sdk-go/tx"
+params := bmTx.NewQueryParamsBuilder().
+    ReferencedBitmark("58737de5ad68a535da6277da62d11eb3ed76ff6dd7fc2adf3c42a4096d9a2518")
+txs, referencedAssets, err := bmTx.List(params)
+```
+
+```java
+TransactionQueryBuilder builder = new TransactionQueryBuilder()
+                                .referencedBitmark("58737de5ad68a535da6277da62d11eb3ed76ff6dd7fc2adf3c42a4096d9a2518");
+Transaction.list(builder, new Callback1<GetTransactionsResponse>() {
+            @Override
+            public void onSuccess(GetTransactionsResponse res) {
+
+            }
+
+            @Override
+            public void onError(Throwable throwable) {
+
+            }
+        });
+```
+
```

Comment:
What returns do we expect when asking for provenance? An ordered history of transactions? For these things returning sets, we obviously don't list out all the elements again, but we still want readers to know what sorts of sets to expect. This is probably the case for all the returns, above and below, that return more than just one element.

Reply:
I think the layout should tell the return is a single record or a set of records.